frontend/src/pages/Students.vue

```diff
@@ -2,294 +2,290 @@
-                        label: 'Date Joined',
-                        searchable: true,
-                    },
-                    {
-                        field: 'StudentID',
-                        label: 'Student ID',
-                        searchable: true,
-                    },
-                    {
-                        field: 'FullName',
-                        label: 'Student Name',
-                        searchable: true,
-                    },                               
-                    {
-                        field: 'StatusID',
-                        label: 'Status',
-                        searchable: true,
-                    },
-                    {
-                        field: 'PhoneNumber',
-                        label: 'Phone Number',
-                        searchable: true,
-                    }
-                ]
-            }
+  data() {
+    return {
+      selected: null,
+      sortIcon: "arrow-up",
+      sortIconSize: "is-small",
+      sortDirection: "asc",
+      columns: [
+        {
+          field: "CreatedAt",
+          label: "Date Joined",
+          searchable: true,
+        },
+        {
+          field: "StudentID",
+          label: "Student ID",
+          searchable: true,
+        },
+        {
+          field: "FullName",
+          label: "Student Name",
+          searchable: true,
+        },
+        {
+          field: "Status",
+          label: "Status",
+          searchable: true,
         },
-        computed:{
-          tableData(){
-            return this.data.map(student => {
-              return {
-                StudentID: `${student.StudentID}`,
-                FullName: `${student.FirstName} ${student.LastName}`,
-                CreatedAt: `${student.created_at}`,
-                StatusID: `${student.StatusID}`,
-                PhoneNumber: `${student.PhoneNumber}`  
-              }                
-            })
-          }
+        {
+          field: "PhoneNumber",
+          label: "Phone Number",
+          searchable: true,
         },
-        components:{
-          Page,
-          PageHeader
-        },        
+      ],
+    };
+  },
+  computed: {
+    tableData() {
+      return this.$store.getters.students.map((student) => {
+        return {
+          StudentID: `${student.StudentID}`,
+          FullName: `${student.FullName}`,
+          CreatedAt: `${student.created_at}`,
+          Status: `${student.status.Description}`,
+          PhoneNumber: `${student.PhoneNumber}`,
+        };
+      });
+    },
+  },
+  components: {
+    Page,
+    PageHeader,
+  },
 
-        // Placeholder for API
-        mounted() {
-          fetch(API_URL)
-            .then(response => response.json())
-            .then(result => {
-              this.data = result;
-            });
-        }
-      
-    }   
+  // Placeholder for API
+  mounted() {
+    this.$store.dispatch("getAllStudents");
```

Comment:
This can be refactored out using the vuex `mapActions` helper https://vuex.vuejs.org/guide/actions.html#dispatching-actions-in-components

Reply:
Thanks! have made the changes